posthog/api/user.py

```diff
@@ -32,16 +33,16 @@ class Meta:
 # TODO: remake these endpoints with DRF!
 @authenticate_secondarily
 def user(request):
-    organization = request.user.organization
+    organization: Optional[Organization] = request.user.organization
     organizations = list(request.user.organizations.order_by("-created_at").values("name", "id"))
-    team = request.user.team
+    team: Optional[Team] = request.user.team
```

Comment:
```
⟩ grep 'request.user.team' **/*.py | wc -l
65
⟩ grep 'request.user.org' **/*.py | wc -l                   
15
```

A lot of these will now raise 500-s when called. This seems like an issue that will start plaguing us long-term.

Thoughts of what we can do here?

Reply:
Added permission class `ProjectMembershipNecessaryPermissions` to endpoints where this could be a problem, which allows for still asserting `request.user.team is not None` and `request.user.organization is not None` in the API.